src/map_data.c

```diff
@@ -142,6 +142,16 @@ void set_navigation_map(MapSubtlCoord stl_x, MapSubtlCoord stl_y, unsigned long
   game.navigation_map[navmap_tile_number(stl_x,stl_y)] = navcolour;
 }
 
+/**
+ * Returns if the position isn't filled with solid block
+ */
+TbBool subtile_not_part_of_solid_block(MapSubtlCoord stl_x, MapSubtlCoord stl_y)
+{
+    struct Map* mapblk = get_map_block_at(stl_x, stl_y);
+    // Check if the position isn't filled with solid block
+    return (((mapblk->flags & SlbAtFlg_Blocking) == 0) && (get_navigation_map_floor_height(stl_x, stl_y) < 4));
```

Comment:
I dont like 4

Reply:
Don't like the value, or the fact that it's a magic number?

Reply:
dont like magic number without some explaination. Is it four cubes or what?

Reply:
Yes, I consider blocks with 4 solid cubes or more to be solid for pathfinding. If I ever get this feature to work properly I'll clarify this 4.